ctru-rs/src/gfx.rs

```diff
@@ -28,29 +28,27 @@ pub trait Screen {
     fn set_framebuffer_format(&mut self, fmt: FramebufferFormat) {
         unsafe { ctru_sys::gfxSetScreenFormat(self.as_raw(), fmt.into()) }
     }
-
-    /// Returns a tuple containing a pointer to the specifified framebuffer (as determined by the
-    /// calling screen and `Side`), the width of the framebuffer in pixels, and the height of
-    /// the framebuffer in pixels
-    ///
-    /// Note that the pointer returned by this function can change after each call to this function
-    /// if double buffering is enabled
-    fn get_raw_framebuffer(&self, side: Side) -> (*mut u8, u16, u16) {
-        let mut width: u16 = 0;
-        let mut height: u16 = 0;
-        unsafe {
-            let buf: *mut u8 =
-                ctru_sys::gfxGetFramebuffer(self.as_raw(), side.into(), &mut width, &mut height);
-            (buf, width, height)
-        }
-    }
 }
 
 #[non_exhaustive]
 pub struct TopScreen;
+
 #[non_exhaustive]
 pub struct BottomScreen;
 
+#[derive(Debug)]
+/// Representation of a framebuffer for one [`Side`] of the top screen, or the
+/// entire bottom screen. The inner pointer is only valid for one frame if double
+/// buffering is enabled. Data written to `ptr` will be rendered to the screen.
+pub struct RawFrameBuffer {
+    /// Pointer to graphics data to be rendered.
+    pub ptr: *mut u8,
```

Comment:
You should give this pointer the `Screen`'s lifetime, otherwise it would be valid after a `Gfx` drop.

Reply:
Something like this? 
```rust
pub struct RawFrameBuffer<'screen> {
    /// Pointer to graphics data to be rendered.
    pub ptr: *mut u8,
    /// The width of the framebuffer in pixels.
    pub width: u16,
    /// The height of the framebuffer in pixels.
    pub height: u16,

    _screen: PhantomData<&'screen mut dyn Screen>,
}
```

Then the impl looks more like 
```rust
impl<'screen> RawFrameBuffer<'screen> {
    fn for_screen_side(screen: &'screen mut dyn Screen, side: ctru_sys::gfx3dSide_t) -> Self {
    }
}
```

This seems to prevent dropping the `Screen` before dropping the `FrameBuffer`, which I suppose is the goal?

Reply:
Yes, this should work.